jetty-http/src/main/java/org/eclipse/jetty/http/HttpTokens.java

```diff
@@ -231,5 +231,60 @@ else if (b >= 0x80) // OBS
             }
         }
     }
+
+    /**
+     * This is used when decoding to not decode illegal characters based on RFC9110.
+     * CR, LF, or NUL are replaced with ' ', all other control and multibyte characters
+     * are replaced with '?'. If this is given a legal character the same value will be returned.
+     *
+     * @param c the character to test.
+     * @return the original character or the replacement character ' ' or '?'.
```

Comment:
Need to say more about the char return.  Specifically that it is known to be ISO-8859-1 character (i.e can have a simple coerce to byte conversion applied to it).